src/cli/cli.cc

```diff
@@ -895,21 +900,28 @@ int main (const int argc, const char* argv[]) {
         }
       }
 
-      auto host = optionValue(arg, "--host");
-      if (host.size() > 0) {
-        devHost = host;
-      } else {
-        if (flagBuildForIOS || flagBuildForAndroid) {
-          auto r = exec("ifconfig | grep -w 'inet' | awk '!match($2, \"^127.\") {print $2; exit}' | tr -d '\n'");
-          if (r.exitCode == 0) {
-            devHost = r.output;
+      if (hostArg.size() == 0) {
+        hostArg = optionValue(arg, "--host");
+        if (hostArg.size() > 0) {
+          devHost = hostArg;
+        } else {
+          if (flagBuildForIOS || flagBuildForAndroid) {
+            auto r = exec((!platform.win) ? 
+              "ifconfig | grep -w 'inet' | awk '!match($2, \"^127.\") {print $2; exit}' | tr -d '\n'" : 
```

Comment:
this will fail on macOS because of the quote escapes in awk's match expression. removing them fixes it.

```bash
ifconfig | grep -w 'inet' | awk '!match($2, "^127.") {print $2; exit}' | tr -d '\n'
```

Reply:
Maybe we could do this instead of spawning so many external programs...

```
#include <stdlib.h>
#include <unistd.h>
#include <netdb.h>
#include <arpa/inet.h>

char* getIP () {
   char host[256];
   struct hostent *host_entry;
   gethostname(host, sizeof(host));
   host_entry = gethostbyname(host);
   return inet_ntoa(*((struct in_addr*) host_entry->h_addr_list[0]));
}
```

Reply:
@heapwolf this is still showing as Changes Requested even though you have resolved it.

Have had a look at replacing the shell commands with libuv as discussed in discord..
The issue is that the shell commands return useful values, because they tend to locate an address with a gateway which is most likely
to be an non internal, non virtual ip address.
I can't see any way with libuv to get an address with a gateway. The first non internal address returned by libuv tends to be just a random virtual interface, or one that isn't connected (eg vpn, ethernet on mac os, WSL on windows)
Based on these two posts, there dooesn't seem to be a clean way to do it in general or using libuv.
https://stackoverflow.com/a/3288983/3739540
https://github.com/nodejs/node/issues/5417